test/unit/webdriver/device/keyboard_test.py

```diff
@@ -32,3 +32,69 @@ def test_hide_keyboard(self):
             appium_command('/session/1234567890/appium/device/hide_keyboard')
         )
         assert isinstance(driver.hide_keyboard(), WebDriver)
+
+    @httpretty.activate
+    def test_press_keycode(self):
+        driver = android_w3c_driver()
+        httpretty.register_uri(
+            httpretty.POST,
+            appium_command('/session/1234567890/appium/device/press_keycode'),
+            body='{"value": "86"}'
+        )
+        driver.press_keycode(86)
+        d = get_httpretty_request_body((httpretty.last_request()))
+        assert d['keycode'] == 86
+
+    @httpretty.activate
+    def test_long_press_keycode(self):
+        driver = android_w3c_driver()
+        httpretty.register_uri(
+            httpretty.POST,
+            appium_command('/session/1234567890/appium/device/long_press_keycode'),
+            body='{"value": "86"}'
+        )
+        driver.long_press_keycode(86)
+        d = get_httpretty_request_body((httpretty.last_request()))
+        assert d['keycode'] == 86
+
+    @httpretty.activate
+    def test_keyevent(self):
+        driver = android_w3c_driver()
+        httpretty.register_uri(
+            httpretty.POST,
+            appium_command('/session/1234567890/appium/device/keyevent'),
+            body='{keycode: 86}'
+        )
+        assert isinstance(driver.keyevent(86), WebDriver)
+
+    @httpretty.activate
+    def test_press_keycode_with_flags(self):
+        driver = android_w3c_driver()
+        httpretty.register_uri(
+            httpretty.POST,
+            appium_command('/session/1234567890/appium/device/press_keycode'),
+            body='{keycode: 86, metastate: 2097153, flags: 44}'
+        )
+        # metastate is META_SHIFT_ON and META_NUM_LOCK_ON
+        # flags is CANCELFLAG_CANCELEDED, FLAG_KEEP_TOUCH_MODE, FLAG_FROM_SYSTEM
+        assert isinstance(
+            driver.press_keycode(
+                86, metastate=[
+                    0x00000001, 0x00200000], flags=[
+                    0x20, 0x00000004, 0x00000008]), WebDriver)
+
```

Comment:
Add check arguments which are passed to appium server.

```suggestion
        d = get_httpretty_request_body((httpretty.last_request()))
        assert d['keycode']  == 86
        assert d['metastate'] == [blabla]
        assert d['flags'] == [blabla]
```